internal/testutils/parse.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"strings"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/protoparse"
+	"github.com/lithammer/dedent"
+)
+
+// ParseProtoString parses a string representing a proto file, and returns
+// a FileDescriptor.
+//
+// It dedents the string before parsing.
+// It is unable to handle imports, and panics if there is any error.
+func ParseProtoString(src string) *desc.FileDescriptor {
```

Comment:
Consider `ParseProtoStringDedented(fileName, src string) *desc.FileDescriptor`:
1. It informs developers that the `src` will be dedented -- note in comments is not enough and they will be surprise when checking the returned locations;
2. `fileName` allows them to specify the wanted file name, instead of the baked `"test.proto"`.

Reply:
1. At least for me, the way I check returned locations is to "get it wrong" (usually by putting in intentionally bogus numbers) and then letting the test fail, and then modifying the test to match the location I actually got. Also, it will be a surprise once, and then after that it is a convenience.

2. YAGNI.

Reply:
+1 to mingzhi's suggestion, but I don't feel too strongly about it for a test utility.

Reply:
I would like to leave this one as-is. It exchanges a one-time surprise for long-time convenience.